tests/Browser/Pages/Backend/Categories/DeleteCategoryTest.php

```diff
@@ -0,0 +1,127 @@
+    }
+
+    /**
+     * Test Admin can't see button delete on default category
+     *
+     * @return void
+     */
+    public function testDontDeleteDefaultCategory()
+    {
+        $this->browse(function (Browser $browser) {
+            $browser->loginAs(User::first())
+                    ->visit('/admin/categories');
+            $btnDelete = $browser->elements('#list-categories tbody tr:nth-child('.Category::CATEGORY_DEFAULT.') button');
+            $this->assertCount(0, $btnDelete);
+        });
+    }
+
+    /**
+     * Test Admin click button delete on List Categories
+     *
+     * @return void
+     */
+    public function testClickButtonDelete()
+    {
+        $this->browse(function (Browser $browser) {
+            $browser->loginAs(User::first())
+                    ->visit('/admin/categories')
+                    ->click('td button.btn-delete-item')
+                    ->assertSee('Confirm deletion!');
+        });
+    }
+
+    /**
+     * Test Admin click button Close on pop up confirmation delete
+     *
+     * @return void
+     */
+    public function testClickButtonCloseOnPopup()
+    {
+        $this->browse(function (Browser $browser) {
+            $browser->loginAs(User::first())
+                    ->visit('/admin/categories')
+                    ->click('td button.btn-delete-item')
+                    ->press('Close')
+                    ->assertDontSee('Confirm deletion!');
+        });
+    }
+
+    /**
+     * Test Admin click button Delete on pop up confirmation delete
+     *
+     * @return void
+     */
+    public function testClickButtonDeleteOnPopup()
+    {
+        $this->browse(function (Browser $browser) {
+            $browser->loginAs(User::first())
+                    ->visit('/admin/categories');
+
+            $totalBooksDefault = $browser->text('#list-categories tbody tr:nth-child('.Category::CATEGORY_DEFAULT.') td:nth-child(3)');
+            $totalBooksDelete = $browser->text('#list-categories tbody tr:nth-child('.self::NO_CATEGORY_DELETE.') td:nth-child(3)');
+            $totalBooksDefaultNew = $totalBooksDefault + $totalBooksDelete;          
+
+            $browser->click('#list-categories tbody tr:nth-child('.self::NO_CATEGORY_DELETE.') button.btn-delete-item')
+                    ->press('Delete')
+                    ->assertDontSee('Confirm deletion!')
+                    ->assertSee('Successfully deleted ');
+            // Count number books of category default
+            $totalBooksDefault = $browser->text('#list-categories tbody tr:nth-child('.Category::CATEGORY_DEFAULT.') td:nth-child(3)');            
+            $this->assertTrue($totalBooksDefault == $totalBooksDefaultNew);                   
+            // Count row number in one page
+            $elements = $browser->elements('#list-categories tbody tr');
+            $this->assertCount(self::NUMBER_RECORD_CATEGORY - 1, $elements);
+        });
+    }
+
```

Comment:
test cho anh trường hợp delete category không tồn tại à trường hợp delete category default nữa em.
muốn test xoá category default em có thể gọit rược triếp route của nó:
$this->jsonAsUser('DELETE', "url/{$id}");